web/shared/src/typeahead.ts

```diff
@@ -105,6 +101,18 @@ export function query_matches_string_in_order(
     query = query.toLowerCase();
     query = remove_diacritics(query);
 
+    return match_query_with_source(query, source_str, split_char);
+}
+
+// This function attempts to match a query in order with a source text.
+// * query is the user-entered search query
+// * source_str is the string we're matching in, e.g. a user's name
+// * split_char is the separator for this syntax (e.g. ' ').
+export function match_query_with_source(
```

Comment:
`query_matches_string_in_order_assume_canonicalized` might be a clearer name for this -- we want to indicate how this relates to the previous.